packages/salesforce/src/bulk1.js

```diff
@@ -0,0 +1,248 @@
+import { connection } from './Adaptor';
+import { composeNextState } from '@openfn/language-common';
+import { expandReferences } from '@openfn/language-common/util';
+
+/**
+ * State object for Salesforce Bulk API 1.0 operations
+ * @typedef {Object} Bulk1QueryState
+ * @property {Array} data - The processed records or query results from the Bulk API operation
+ * @property {Array} references - Array of previous state data objects used in the job
+ * @private
+ */
+
+/**
+ * @typedef {Object} Bulk1LoadState - State object for bulk load operations
+ * @property {Object} data - The processed records or results from the bulk load operation
+ * @property {Object} data.successfulResults - Array of successful results
+ * @property {Object} data.failedResults - Array of failed results
+ * @property {Object} data.unprocessedRecords - Array of unprocessed records
+ * @property {Array} references - Array of previous state data objects used in the job
+ * @private
+ */
+
+/**
+ * Options for configuring Salesforce Bulk API 1.0 operations
+ * @typedef {Object} Bulk1Options
+ * @public
+ * @property {string} [extIdField] - External id field. Required for upsert operations
+ * @property {boolean} [allowNoOp=false] - Skipping bulk operation if no records
+ * @property {boolean} [failOnError=false] - Fail the operation on error
+ * @property {number} [pollTimeout=240000] - Polling timeout in milliseconds. Default: 240000 (4 minutes)
+ * @property {number} [pollInterval=6000] - Polling interval in milliseconds. Default: 6000 (6 seconds)
+ * @property {boolean} [concurrencyMode='Parallel'] - Concurrency mode: 'Parallel' or 'Serial'
+ */
+
+/**
+ * Executes a bulk query using Salesforce Bulk API 1.0
+ * @public
+ * @example <caption>Query records</caption>
+ * bulk1.query('SELECT Id, Name FROM Account');
+ * @example <caption>Query with custom options</caption>
+ * bulk1.query('SELECT Id, Name FROM Account', {
+ *   pollInterval: 1000,
+ *   pollTimeout: 30000
+ * });
+ * @function
+ * @param {string} query - SOQL query string to execute
```

Comment:
Should it be `SQL` or `SOQL`?

Reply:
@hunterachieng it's SOQL stands for Salesforce Object Query Language, [Learn more here](https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_functions.htm)